### PR TITLE
fix(template): unescape escape templates when not doing partial resolution

### PR DESCRIPTION
**Pull request by edvald**
Fixes: #3989

### CODE REVIEW

**Review by eysi09 — DISMISSED** [review]
LGTM

**Comment by TimBeyer** [comment]
When I was trying to resolve this I had a few more test cases. Maybe it makes sense to check if those pass too?

```
  it("should allow nesting normal strings within escaped strings", () => {
    const res = resolveTemplateString("$${var.${foo}}", new TestContext({ foo: "yes" }), { unescape: true })
    expect(res).to.equal("${var.yes}")
  })

  it("should allow nesting escaped strings within normal strings", () => {
    const res = resolveTemplateString("${foo == 'yes' ? '$${bar}' : 'fail' }", new TestContext({ foo: "yes" }), { unescape: true })
    expect(res).to.equal("${bar}")
  })


  it.only("should escape things correctly", () => {
    const res = resolveTemplateString("$${env.TEST_ENV}", new TestContext({}), {
      unescape: true,
    })
    expect(res).to.equal("${env.TEST_ENV}")
  })

  it.only("should escape things correctly", () => {
    const res = resolveTemplateString("foo $${env.TEST_ENV} bar", new TestContext({}), {
      unescape: true,
    })
    expect(res).to.equal("foo ${env.TEST_ENV} bar")
  })

  it.only("should escape things correctly", () => {
    const res = resolveTemplateString("$${env:TEST_ENV}", new TestContext({}), {
      unescape: true,
    })
    expect(res).to.equal("${env:TEST_ENV}")
  })


  it.only("should escape things correctly", () => {
    const res = resolveTemplateString("foo $${env:TEST_ENV} bar", new TestContext({}), {
      unescape: true,
    })
    expect(res).to.equal("foo ${env:TEST_ENV} bar")
  })

  it("should escape things correctly", () => {
    const res = resolveTemplateString("${foo}-$${env:TEST_ENV}", new TestContext({ foo: "foo" }), {
      unescape: true,
    })
    expect(res).to.equal("foo-${env:TEST_ENV}")
  })
```

**Comment by eysi09** [comment]
@TimBeyer, yeah good catch. 

Also, the framework tests appear to be failling.

File: core/test/unit/src/template-string.ts [hunk 1]
```diff
@@ -71,8 +71,12 @@ describe("resolveTemplateString", () => {
     expect(res).to.equal("${bar}")
   })
 
-  it("should pass through a template string with a double $$ prefix", () => {
-    const res = resolveTemplateString({ string: "$${bar}", context: new TestContext({}) })
+  it("should pass through a template string with a double $$ prefix if allowPartial=true", () => {
+    const res = resolveTemplateString({
+      string: "$${bar}",
+      context: new TestContext({}),
+      contextOpts: { allowPartial: true },
```
Comment (stefreak) [inline core/test/unit/src/template-string.ts:78]: It's not really obvious to me how this fixes the end user issue – can we also add an integration test that shows how this fixes the issue in a less synthetic scenario?

**Comment by edvald** [comment]
All fair points, flagging as WIP for now.

Btw the microk8s test failures seem unrelated. Not sure if we've been seeing those as flaky?

**Comment by edvald** [comment]
@TimBeyer this isn't the current behavior, so it would be a breaking change:

> it("should allow nesting normal strings within escaped strings", () => {
    const res = resolveTemplateString("$${var.${foo}}", new TestContext({ foo: "yes" }), { unescape: true })
    expect(res).to.equal("${var.yes}")
  })

**Review by eysi09 — APPROVED** [review]

**Pull request merged**